vscode/microsoft-kiota/src/extension.ts

```diff
@@ -289,29 +293,10 @@ export async function activate(
         }
       }
     ),
-    registerCommandWithTelemetry(reporter, `${treeViewId}.closeDescription`, async () => {
-      const yesAnswer = vscode.l10n.t("Yes");
-      const response = await vscode.window.showInformationMessage(
-        vscode.l10n.t("Do you want to remove this API description?"),
-        yesAnswer,
-        vscode.l10n.t("No")
-      );
-      if (response === yesAnswer) {
-        openApiTreeProvider.closeDescription();
-        await updateTreeViewIcons(treeViewId, false);
-      }
-    }
-    ),
+    registerCommandWithTelemetry(reporter, closeDescriptionCommand.getName(), async () => await closeDescriptionCommand.execute()),
     registerCommandWithTelemetry(reporter, filterDescriptionCommand.getName(), async () => await filterDescriptionCommand.execute()),
+    registerCommandWithTelemetry(reporter, editPathsCommand.getName(), async () => await editPathsCommand.execute()),
 
-    registerCommandWithTelemetry(reporter, `${extensionId}.editPaths`, async (clientKey: string, clientObject: ClientOrPluginProperties, generationType: string) => {
```

Comment:
this one is going to be problematic:
1. read the values (clientPluginOrKey, etc..) when the extension starts.
2. set copies as properties of the command.
3. use edits the values and triggers the command.
4. the command is still running off old copies of the values.

The bottom line: the command should take in callbacks to read the values when it's triggered instead of copying the values when initializing itself.

Reply:
@baywet why do we copy the properties?

Reply:
we don't, this is just how the runtime works. Scalar properties are passed by value copy, not by reference in javascript.
This ends up being a consequence of your refactoring.

Reply:
I see what you mean. I've changed it to reflect how it was